js/src/forum/components/DiscussionsSearchSource.tsx

```diff
@@ -0,0 +1,58 @@
+import highlight from '../../common/helpers/highlight';
+import LinkButton from '../../common/components/LinkButton';
+import Link from '../../common/components/Link';
+import SearchSource from './SearchSource';
+import Mithril from 'mithril';
+
+/**
+ * The `DiscussionsSearchSource` finds and displays discussion search results in
+ * the search dropdown.
+ */
+export default class DiscussionsSearchSource implements SearchSource {
+  protected results = new Map<string, unknown[]>();
+
+  search(query: string) {
+    query = query.toLowerCase();
+
+    this.results.set(query, []);
+
+    const params = {
+      filter: { q: query },
+      page: { limit: 3 },
+      include: 'mostRelevantPost',
+    };
+
+    return app.store.find('discussions', params).then((results) => this.results.set(query, results));
+  }
```

Comment:
That wouldn't be any changes other than just the modifier, right?

Reply:
It would mean using `await` instead of `.then`.

E.g.

```js
fetch("example.com").then(response => doSomething(response)
```

Would become

```js
const response = await fetch("example.com")
doSomething(response)
```